### PR TITLE
feat(pubsub): Trigger pipeline matches on expectedArtifact IDs.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

**Comment by jtk54** [comment]
Depends on https://github.com/spinnaker/deck/pull/4349 and https://github.com/spinnaker/orca/pull/1763.

File: echo-model/src/main/java/com/netflix/spinnaker/echo/model/Trigger.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public String toString() {
    */
   String subscriptionName;
   String pubsubSystem;
-  List<ExpectedArtifact> expectedArtifacts;
+  List<String> expectedArtifacts;
```
Comment (lwander) [inline echo-model/src/main/java/com/netflix/spinnaker/echo/model/Trigger.java]: Should we call these `expectedArtifactIds`?

Comment (jtk54) [reply echo-model/src/main/java/com/netflix/spinnaker/echo/model/Trigger.java]: Yeah, let me rename.

Comment (jtk54) [reply echo-model/src/main/java/com/netflix/spinnaker/echo/model/Trigger.java]: Changed this everywhere and re-tested, should be ok now.

**Pull request merged**